src/app/persistence/String.h

```diff
@@ -0,0 +1,122 @@
+/**
+ *    Copyright (c) 2025 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <app/persistence/AttributePersistenceProvider.h>
+#include <app/persistence/PascalString.h>
+#include <lib/support/Span.h>
+
+#include <cstddef>
+#include <cstdint>
+
+namespace chip::app::Storage {
+
+namespace Internal {
+
+/// Represents a string that can be read/written to storage.
+///
+/// This is currently implemented for SHORT strings only (< 255 bytes)
+/// that are NOT nullable
+class ShortString
+{
+public:
+    /// Use the input buffer as a short pascal string.
+    ///
+    /// The input buffer is assumed to have:
+    ///   - size for a PascalString (i.e. generally pascal string length + prefix == length + 1 given we use
+    ///     short pascal strings here)
+    ///   - an extra +1 bytes for a null terminator so that c_str works.
+    ///
+    /// This means that buffer_size should be `N+2` where N is the expected maximum string length.
+    ///
+    /// This class is considered Internal, use chip::app::Storage::String<MAX_LENGTH> in application code.
+    ShortString(char * buffer, size_t buffer_size) : mBuffer(buffer), mPascalSize(static_cast<uint8_t>(buffer_size - 1))
+    {
+        // for a buffer to be usable we need 1 byte for size, 1 byte for content and 1 byte for null terminator.
+        // Strings without any size make no sense.
+        VerifyOrDie(buffer_size >= 3);
+        VerifyOrDie(buffer_size < 256);
```

Comment:
This looks wrong.  A string of max length 254 (totally valid for a short string) needs a buffer_size of 256, no?